src/cockpit/samples.py

```diff
@@ -301,6 +301,60 @@ def sample(self, samples: Samples) -> None:
                     samples['cgroup.cpu.usage'][cgroup] = usage_nsec / 1000000
 
 
+class CGroupDiskIO(Sampler):
+    descriptions = [
+        SampleDescription('disk.cgroup.read', 'bytes', 'counter', True),
+        SampleDescription('disk.cgroup.written', 'bytes', 'counter', True),
+    ]
+
+    @staticmethod
+    def get_cgroup_name(fd: int) -> Optional[str]:
+        cgroup_name = ""
+        try:
+            with Handle.open('cgroup', os.O_RDONLY, dir_fd=fd) as cgroup_fd:
+                cgroup_name = os.read(cgroup_fd, 1024).decode().strip()
+        except FileNotFoundError:
```

Comment:
...and stop catching exceptions.  They can bubble out in case the file isn't there.  We already catch exceptions in the toplevel function and do the right thing in that case (skip to the next iteration).

Reply:
...it's pretty frustrating that stock `open()` doesn't take `dir_fd`.  Grr.